File: Python/Gate Four/milky_doughnut.py
def count_even_numbers(numbers):
    even_count = 0

    for number in numbers:
        if number % 2 == 0:
            even_count += 1

    return even_count


def linear_search(numbers, target_value):
    for index in range(len(numbers)):
        if numbers[index] == target_value:
            return index
    return -1


def generate_squares(size):
    squares = []

    for index in range(size):
        number = index + 1
        squares.append(number * number)

    return squares


def are_arrays_equal(first_array, second_array):
    if len(first_array) != len(second_array):
        return False

    for index in range(len(first_array)):
        if first_array[index] != second_array[index]:
            return False

    return True

File: Java/Gate Four/MilkyDoughnut.java
public class MilkyDoughnut {


    public static int countEvenNumbers(int[] numbers) {
        int evenCount = 0;

        for (int index = 0; index < numbers.length; index++) {
            if (numbers[index] % 2 == 0) {
                evenCount++;
            }
        }

        return evenCount;
    }


    public static int linearSearch(int[] numbers, int targetValue) {
        for (int index = 0; index < numbers.length; index++) {
            if (numbers[index] == targetValue) {
                return index;
            }
        }
        return -1;
    }

 
    public static int[] generateSquares(int size) {
        int[] squaresArray = new int[size];

        for (int index = 0; index < size; index++) {
            squaresArray[index] = (index + 1) * (index + 1);
        }

        return squaresArray;
    }


    public static boolean areArraysEqual(int[] firstArray, int[] secondArray) {
        if (firstArray.length != secondArray.length) {
            return false;
        }

        for (int index = 0; index < firstArray.length; index++) {
            if (firstArray[index] != secondArray[index]) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {

        int[] numbers = {1, 2, 3, 4, 5, 6};

        int evenResult = countEvenNumbers(numbers);
        System.out.println("Even numbers count: " + evenResult);

        int searchResult = linearSearch(numbers, 4);
        System.out.println("Index of 4: " + searchResult);

       
        int[] squares = generateSquares(5);
        System.out.print("Squares: ");
        for (int i = 0; i < squares.length; i++) {
            System.out.print(squares[i] + " ");
        }
        System.out.println();

   
        int[] array1 = {1, 2, 3};
        int[] array2 = {1, 2, 3};

        boolean isEqual = areArraysEqual(array1, array2);
        System.out.println("Arrays are equal: " + isEqual);
    }
}
